### ISSUE TITLE
OnMapReadyCallback not called

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#4439]
```java
        mapView = new MapView(this, options);
        ViewGroup viewGroup = (ViewGroup) findViewById(R.id.parentLayout);
        viewGroup.addView(mapView);

        mapView.onCreate(savedInstanceState);
        mapView.getMapAsync(this);
```

When creating a mapview as shown above, I'm not receiving the map async callback. 
Only when I zoom out a bit I'm receiving the callback.

An example, the maps is shown but callback is not invoked:

![device-2016-03-23-113451](https://cloud.githubusercontent.com/assets/2151639/13982635/78cc1154-f0eb-11e5-88fc-e8bffd6eaa3f.png)

Only when you interact (zoom/pan) with the map, the callback is invoked and content is shown:

![device-2016-03-23-113538](https://cloud.githubusercontent.com/assets/2151639/13982658/919379a2-f0eb-11e5-919c-cff3245ebd5d.png)

We need some kind of invalidation or some sort in onStart/onResume to trigger the update?

**Comment by tobrun** [comment]
cc @bleege @zugaldia

**Comment by tobrun** [comment]
Noticing that this is also true for MapFragment, which makes it more important to fix for `4.0.0`

**Comment by tobrun** [comment]
This is a bit strange because the configuration currently setup for MapFragmentActivity works as designed, but others don't... 

**Update** when I call a method that invalidates the map this issue is resolved.

**Comment by tobrun** [comment]
Not really found a solution for this but this resolves it:

```java
   options.minZoom(1);
   options.maxZoom(18);
```

Not having this will not call the callback.

Going to dig deeper in the mapchange events and see what I can find out.

**Comment by tobrun** [comment]
To make it a bit more visual, here is a clip showing the problem:
 - https://www.dropbox.com/s/fmheyvckbp184bv/device-2016-03-23-143103.mp4?dl=0

As you can see we are not receiving a map fully loaded until we load a new content by zoom/pan.

On the java side we receive map change callbacks, when we just open the screen without interaction:

```java
03-23 13:42:12.162 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _0
03-23 13:42:12.162 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _3
03-23 13:42:12.163 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _5
03-23 13:42:12.206 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _0
03-23 13:42:12.206 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _3
03-23 13:42:12.597 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _11
03-23 13:42:12.597 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _8
03-23 13:42:13.326 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _9
03-23 13:42:13.336 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _8
03-23 13:42:13.419 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _9
03-23 13:42:13.422 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _8
03-23 13:42:13.446 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _9
03-23 13:42:13.462 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _8
03-23 13:42:13.475 19327-19327/com.mapbox.mapboxsdk.testapp V/MAPBOXCHANGE: CHANGE _9
```

**Comment by tobrun** [comment]
Note that I'm able to reproduce this on master, but fail at this point to do on release branch. Users have reported this was an issue with snapshot and snapshots are currently build from release branch.

**Comment by tobrun** [comment]
I was able to create a clear video of the issue:

https://www.dropbox.com/s/md7berelecklw2b/device-2016-03-23-220634.mp4?dl=0

First attempt was a works as designed, second attempt is not rendering all the things needed and thus not giving us the callback we need.

**Comment by tobrun** [comment]
Here is a similar example:
https://www.dropbox.com/s/fmheyvckbp184bv/device-2016-03-23-143103.mp4?dl=0

**Comment by tobrun** [comment]
I found out this is only producible on Master and that I thought were related isssues are actually not. Removing this from milestone 4.0.0. 

@brunoabinader would you be able to look into this? I think it's related to on of the refactors that happened after branching away from `master` for `release-ios-3.2.0-android-4.0.0`

**Comment by tobrun** [comment]
We were able to git bisect the issue to:

```
Tobruns-Mac-mini:mapbox-gl-native nurbot$ git bisect bad
cd1a06c2dc209da81b3d745c088e568b3b14809f is the first bad commit
commit cd1a06c2dc209da81b3d745c088e568b3b14809f
Author: Bruno de Oliveira Abinader <bruno@mapbox.com>
Date:   Fri Mar 11 14:57:12 2016 +0200

    [core] Get rid of MapData::{get,set}animationTime

:040000 040000 efe082eda3375b2fa6e0fe6848ab4bd8f1b135ba c4f8b9da6827cf7425a4ca5b7fbec74074597765 M	src
:040000 040000 97e27d17a2390e96f96f959a5c5d5eb76e98abd1 cf7fb350ccb2d52a54d956f42c16bdce366e0d15 M	test
```

**Comment by LouisCAD** [comment]
I'm getting this issue in version 5.4.0

**Comment by LouisCAD** [comment]
This seems to happen when the visibility of the container of the `SupportMapFragment` is set to `View.GONE`.

**Comment by zugaldia** [comment]
@LouisCAD this problem was present in v4.x, could you cut a [separate ticket](https://github.com/mapbox/mapbox-gl-native/issues/new) to track your issue? Thanks!

**Comment by tobrun** [comment]
@LouisCAD starting of 5.2.0 we need to the hosting view to be measured before we start loading the map resources and thus invoking the OnMapReady callback. This is expected behaviour.
